### DISCUSSION TITLE
Doubts about using getInitialProps in _app.jsx

### DISCUSSION THREAD

**Discussion by iiDestiny** [vercel/next.js#37030] [Help]
I have a Nextjs question for guys, now the official recommendation is to use getServerSideProps and getStaticProps, but _app.jsx needs to getInitialProps to perform some global operations "such as validate token, determine isMobile on the server side, redux -Saga, etc." and pass it to each page, but then I can't use getServerSideProps.

have any suggestions?

```
const NextApp = ({ Component, pageProps }) => {

  return (
    <>
      <Component {...pageProps} />
    </>
  )
}

NextApp.getInitialProps = wrapper.getInitialAppProps(
  (store) =>
    async ({ Component, ctx }) => {
      let pageProps = {}
      let isMobile = false

      if (isClient()) {
        isMobile = clientIsMobile()
      } else {
        isMobile = !!ssrIsMobile(ctx.req)
      }

      // auth
      if (Component.auth && Component.auth() && !authCheck(ctx)) {
        console.log('Auth protection')
        clearUserDataTo(store, ctx)
      }

      // guest 
      if (Component.guest && authCheck(ctx)) {
        guestRedirect(ctx)
      }

      // initNextServer
      await initNextServer(ctx, store)


      if (Component.getInitialProps) {
        try {
          pageProps = await Component.getInitialProps({
            ...ctx,
            store,
            isMobile,
          })
        } catch (e) {
          console.log('_app:', e?.response?.message)
          const statusCode = e?.response?.status
          // 401 
          if (!isClient() && statusCode === 401) {
            clearUserDataTo(store, ctx)
          }
        }
      }

      if (ctx.req) {
        console.log('Saga is executing on server, we will wait')
        store.dispatch(END)
        await store.sagaTask.toPromise()
        console.log('Saga is executing on server, done')
      }

      return {
        pageProps: {...pageProps, isMobile },
      }
    }
)

export default appWithTranslation(wrapper.withRedux(NextApp))
```

**Comment by icyJoseph** [comment]
Why wouldn't you be able to run GSSP?

> However, if getInitialProps is used in a custom _app.js, and the page being navigated to implements getServerSideProps, then getInitialProps will run on the server.

Is this the problem? I just doubled checked with a toy project, and `getInitialProps` on `pages/_app.tsx` with `GSSP` on `pages/blog.tsx` worked fine. What problems are you seeing?

**Reply by iiDestiny** [reply]
My question is this.
. global data is shared to each page using `getInitialProps` in _app.jsx "if there is an alternative to getInitialProps"

 is it possible to use `getServerSideProps ` to receive this shared data in a separate page?

I'm using getInitialProps for all pages now, because it can receive data from _app.jsx. I'm confused, if I use getServerSideProps then how do I ensure global data reception?

![image](https://user-images.githubusercontent.com/17637481/169245919-2a5638f6-94e7-48e9-a262-e6ab4d9ce6d2.png)
